packages/features/bookings/lib/handleNewBooking/loadUsers.ts

```diff
@@ -0,0 +1,102 @@
+import { Prisma } from "@prisma/client";
+import type { IncomingMessage } from "http";
+
+import { orgDomainConfig } from "@calcom/features/ee/organizations/lib/orgDomains";
+import { HttpError } from "@calcom/lib/http-error";
+import logger from "@calcom/lib/logger";
+import { UserRepository } from "@calcom/lib/server/repository/user";
+import prisma, { userSelect } from "@calcom/prisma";
+import { credentialForCalendarServiceSelect } from "@calcom/prisma/selects/credential";
+
+import type { NewBookingEventType } from "../handleNewBooking";
+
+const log = logger.getSubLogger({ prefix: ["[loadUsers]:handleNewBooking "] });
+type Hosts = NewBookingEventType["hosts"];
+
+export const loadUsers = async (
+  eventType: NewBookingEventType,
+  dynamicUserList: string[],
+  req: IncomingMessage
+) => {
+  try {
+    return eventType.id
+      ? await loadUsersByEventType(eventType)
+      : await loadDynamicUsers(dynamicUserList, req);
+  } catch (error) {
+    if (error instanceof HttpError || error instanceof Prisma.PrismaClientKnownRequestError) {
+      throw new HttpError({ statusCode: 400, message: error.message });
+    }
+    throw new HttpError({ statusCode: 500, message: "Unable to load users" });
+  }
+};
+
+const loadUsersByEventType = async (
+  eventType: NewBookingEventType
+): Promise<NewBookingEventType["users"]> => {
+  const hosts = eventType.hosts || [];
+  validateHosts(hosts);
+  const users = hosts.map(({ user, isFixed, priority }) => ({
+    ...user,
+    isFixed,
+    priority,
+  }));
+  return users.length ? users : eventType.users;
+};
+
+const validateHosts = (hosts: Hosts): void => {
+  if (!Array.isArray(hosts)) {
+    throw new Error("eventType.hosts is not properly defined.");
+  }
+};
+
+const loadDynamicUsers = async (dynamicUserList: string[], req: IncomingMessage) => {
+  validateDynamicUserList(dynamicUserList);
+  const { isValidOrgDomain, currentOrgDomain } = orgDomainConfig(req);
+  return findUsersByUsername({
+    usernameList: dynamicUserList,
+    orgSlug: isValidOrgDomain ? currentOrgDomain : null,
+  });
+};
+
+const validateDynamicUserList = (dynamicUserList: string[]): void => {
+  if (!Array.isArray(dynamicUserList) || dynamicUserList.length === 0) {
+    throw new Error("dynamicUserList is not properly defined or empty.");
+  }
+};
+
+/**
+ * This method is mostly same as the one in UserRepository but it includes a lot more relations which are specific requirement here
+ * TODO: Figure out how to keep it in UserRepository and use it here
+ */
+export const findUsersByUsername = async ({
```

Comment:
no change in the code here.

Reply:
this function returns few more fields(credentials and metadata) than the static function in UserRepository.